examples/taskworkflow.go

```diff
@@ -56,30 +59,32 @@ func main() {
 
 func taskWorkflow(containerIP string, gateway string, netMask string) (err error) {
 	log.Println("Creating containerd client")
-	client, err := containerd.New("/run/containerd/containerd.sock")
+	client, err := containerd.New(containerdAddress)
 	if err != nil {
 		return errors.Wrapf(err, "creating client")
 
 	}
 	defer client.Close()
 	log.Println("Created containerd client")
 
-	ctx := namespaces.WithNamespace(context.Background(), "firecracker-containerd-example")
+	ctx := namespaces.WithNamespace(context.Background(), namespaceName)
 	image, err := client.Pull(ctx, "docker.io/library/nginx:latest",
 		containerd.WithPullUnpack,
 		containerd.WithPullSnapshotter("firecracker-naive"),
 	)
 	if err != nil {
 		return errors.Wrapf(err, "creating container")
+	}
 
+	pluginClient, err := ttrpcutil.NewClient(containerdAddress + ".ttrpc")
+	if err != nil {
+		return errors.Wrap(err, "failed to create ttrpc client")
 	}
 
-	fcClient := fccontrol.NewFirecrackerClient(client.Conn())
+	fcClient := fccontrol.NewFirecrackerClient(pluginClient.Client())
```

Comment:
Would it be possible to change `NewFirecrackerClient` to accept the ttrpc socket path instead of a pre-existing ttrpc client? So it would be like `fcClient := fccontrol.NewFirecrackerClient(containerdAddress + ".ttrpc")`?

That's will make things a little simpler for users of the client who aren't familiar with how it works. We could also have a separate `NewFirecrackerClientFrom(existingTTRPCClient)` to maintain the existing interface in case that's ever needed too (though I'm not sure it is in our codebase at this time).

Reply:
The client is [auto generated](https://github.com/containerd/ttrpc/blob/a5bd8ce9e40bc7c065a11c6936f4d032ce6bfa2b/plugin/generator.go#L111) by ttrpc. While I see your point, I don't think it worth changing it manually or submitting a PR with breaking changes.

Reply:
Ah right, can we add a util function somewhere else then? It just doesn't make sense to force users to always jump through these multiple hoops in order to create a client when they are going to be doing the same thing pretty much every time.

Reply:
Discussed privately, will add a client wrapper helper in a separate PR.